security/s2a/internal/crypter/ciphersuite_test.go

```diff
@@ -0,0 +1,32 @@
+package crypter
+
+import (
+	s2a_proto "google.golang.org/grpc/security/s2a/internal/proto"
+	"testing"
+)
+
+func TestCiphersuites(t *testing.T) {
+	for _, tc := range []struct {
+		s2aProtoCiphersuite s2a_proto.Ciphersuite
+		expectedCiphersuite ciphersuite
+	}{
+		{
+			s2aProtoCiphersuite: s2a_proto.Ciphersuite_AES_128_GCM_SHA256,
+			expectedCiphersuite: &aesgcm128sha256{},
+		},
+		{
+			s2aProtoCiphersuite: s2a_proto.Ciphersuite_AES_256_GCM_SHA384,
+			expectedCiphersuite: &aesgcm256sha384{},
+		},
+		{
+			s2aProtoCiphersuite: s2a_proto.Ciphersuite_CHACHA20_POLY1305_SHA256,
+			expectedCiphersuite: &chachapolysha256{},
+		},
+	} {
+		t.Run(tc.s2aProtoCiphersuite.String(), func(t *testing.T) {
+			if got, want := newCiphersuite(tc.s2aProtoCiphersuite), tc.expectedCiphersuite; got != want {
```

Comment:
I think we need to improve the test coverage a bit. One way to do this would be to check that `tc.s2aProtoCiphersuite.keySize()` is equal to what it should be, and so on for the other methods.

For the AEAD crypter, there's not much we can check, but perhaps we can ensure it is an instance of the correct type? (For example, in Java, the test condition would be `aeadCrypter instanceof aesgcm` or similar. I'm not sure whether Go has something similar.) Do you know if there's an easy way to do this in Go?

Expanding the coverage may be a bit repetitive, but this test's goal is to tell us if anything in `ciphersuite.go` changes (and not really to test the current implementation).